File: Busca/buscaBinaria.c
#include <iostream>
#include <vector>

using namespace std;


int binary_search(vector<int>  v, int x){
    int esquerda = 0;
    int direita = v.size() - 1;
    int meio;
    while (esquerda <= direita){
        meio = (direita - esquerda)/2;
        if (x == v[meio])
            return meio;
        if (x < v[meio])
            direita = meio - 1;
        else
            esquerda = meio + 1;
    }
    return -1;
}



int main(){
    vector<int> v;
    v = {2,3,5,7,11,13,17,19};
    cout << binary_search(v,17);
    
    return 0;
}

File: Busca/buscaBinaria.py

def buscaBinaria(vetor,procurado):
    esquerda = 0
    direita = len(vetor) - 1
    while esquerda <= direita:
        meio = (esquerda + direita)//2
        if procurado == vetor[meio]:
            return meio
        if procurado < vetor[meio]:
            direita = meio - 1
        else:
            esquerda = meio + 1 
    
    return None


print(buscaBinaria([2,3,5,7,11,13,17],5))
print(buscaBinaria([2,3,5,7,11,13,17],17))
print(buscaBinaria([2,3,5,7,11,13,17],19))
